### PR TITLE
bridge: add caller support in contract call

### PR DESCRIPTION
**Pull request by chenfengjin**
## Description

add caller support in contract call

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: kernel/contract/mock/helper.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type TestHelper struct {
 }
 
 func NewTestHelper(cfg *contract.ContractConfig) *TestHelper {
-	basedir, err := ioutil.TempDir("", "contract-test")
+	basedir, err := ioutil.TempDir("/tmp", "contract-test")
```
Comment (icexin) [inline kernel/contract/mock/helper.go:34]: 第一个参数留空会自动在临时目录创建，不同系统的临时目录地方不一样

Comment (chenfengjin) [reply kernel/contract/mock/helper.go]: 已修复

File: kernel/contract/manager/testdata/go.mod [hunk 2]
```diff
@@ -3,3 +3,5 @@ module contract-test
 go 1.15
 
 require github.com/xuperchain/contract-sdk-go v0.0.0-20210118130759-2897ea0f985a
+
+replace github.com/xuperchain/contract-sdk-go => /Users/chenfengjin/baidu/contract-sdk-go
```
Comment (qizheng09) [inline kernel/contract/manager/testdata/go.mod:7]: 不要将开发状态的代码提交上来，如果涉及多个仓库，辛苦按照依赖依次提交

Comment (chenfengjin) [reply kernel/contract/manager/testdata/go.mod]: 这个是有什么特殊的考虑吗? 

看了下其他社区，用 WIP 提交开发状态的代码还蛮普遍的
kubernetes： https://github.com/kubernetes/kubernetes/pulls?q=is%3Apr+is%3Aopen+WIP
tidb: https://github.com/pingcap/tidb/pulls?q=is%3Apr+is%3Aopen+WIP
docker: https://github.com/moby/moby/pulls?q=is%3Apr+is%3Aopen+WIP
nodejs:  https://github.com/nodejs/node/pulls?q=is%3Apr+is%3Aopen+WIP

Comment (chenfengjin) [reply kernel/contract/manager/testdata/go.mod]: 这个 PR 也已经 Ready for Review 了

**Review by zheng-bin — APPROVED** [review]
lgtm

**Review by godeamon — APPROVED** [review]

**Review by icexin — APPROVED** [review]

**Pull request merged**